files/build_templates/buffers_config.j2

```diff
@@ -47,10 +48,14 @@ def
             {%- if DEVICE_NEIGHBOR_METADATA is defined and DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[local_port].name] %}
                 {%- set neighbor = DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[local_port].name] %}
                 {%- set neighbor_role = neighbor.type %}
-                {%- set roles1 = switch_role + '_' + neighbor_role %}
-                {%- set roles2 = neighbor_role + '_' + switch_role %}
-                {%- set roles1 = roles1 | lower %}
-                {%- set roles2 = roles2 | lower %}
+                {%- if 'asic' == neighbor_role | lower %}
+                         {%- set roles1 = 'internal' %}
```

Comment:
do we have unit test to cover this change in sonic-config-engine?

Reply:
Added unit test.